src/decode.rs

```diff
@@ -145,13 +145,22 @@ pub unsafe fn hex_check_sse(mut src: &[u8]) -> bool {
     hex_check_fallback(src)
 }
 
+/// Hex decode src into dst.
+/// The length of src must be even and not zero.
+/// The length of dst must be at least src.len() / 2.
 pub fn hex_decode(src: &[u8], dst: &mut [u8]) -> Result<(), Error> {
     if src.is_empty() {
         return Err(Error::InvalidLength(0));
     }
-    let len = dst.len().checked_mul(2).unwrap();
-    if src.len() < len || ((src.len() & 1) != 0) {
-        return Err(Error::InvalidLength(len));
+
+    if src.len() & 1 != 0 {
+        return Err(Error::InvalidLength(src.len()));
+    }
+
+    let expect_dst_len = src.len().checked_div(2).unwrap();
```

Comment:
`usize::checked_div(rhs)` only return `None` if `rhs` is `0`, so it's OK to use `.unwrap()` here.